src/faultlog/util.cpp

```diff
@@ -182,4 +184,58 @@ json parseCallout(const std::string callout)
     return sectionJson;
 }
 
+bool isECOModeEnabled(struct pdbg_target* coreTgt)
+{
+    ATTR_ECO_MODE_Type ecoMode;
+    if (DT_GET_PROP(ATTR_ECO_MODE, coreTgt, ecoMode))
```

Comment:
I guess these two if statements can be combined into one here like
`if (DT_GET_PROP(ATTR_ECO_MODE, coreTgt, ecoMode) || (ecoMode != ENUM_ATTR_ECO_MODE_ENABLED)) return false;
 otherwise return true;`

Reply:
Reason been in the first condition ecoMode would be populated if it is successful and then in the second condition can be evaluated